borg/hashindex.pyx

```diff
@@ -110,22 +110,23 @@ cdef class NSIndex(IndexBase):
 
     def __getitem__(self, key):
         assert len(key) == self.key_size
-        data = <int32_t *>hashindex_get(self.index, <char *>key)
+        data = <uint32_t *>hashindex_get(self.index, <char *>key)
         if not data:
             raise KeyError(key)
         return _le32toh(data[0]), _le32toh(data[1])
 
     def __setitem__(self, key, value):
         assert len(key) == self.key_size
-        cdef int32_t[2] data
+        cdef uint32_t[2] data
+        assert value[0] <= MAX_REF, "maximum number of segments reached"
```

Comment:
in that case, it is a segment number, not a refcount, so MAX_SEGMENT?
or just have a generic MAX_VALUE?